### PR TITLE
directly Wfact

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/927

**Comment by ChrisRackauckas** [comment]
that's required to get finished for this to work.

File: src/DiffEqGPU.jl [hunk 1]
```diff
@@ -154,12 +150,15 @@ function batch_solve(ensembleprob,alg,ensemblealg,I;kwargs...)
     @assert !isempty(I)
     #@assert all(p->p.f === probs[1].f,probs)
 
+    len = length(probs[1].u0)
     if ensemblealg isa EnsembleGPUArray
         u0 = CuArray(hcat([probs[i].u0 for i in 1:length(I)]...))
         p  = CuArray(hcat([probs[i].p  for i in 1:length(I)]...))
+        jac_prototype = cu(zeros(Float32,len,len,length(I)))
```
Comment (ChrisRackauckas) [inline src/DiffEqGPU.jl]: much nicer

**Comment by YingboMa** [comment]
Tests pass locally, so I am going to merge after https://github.com/JuliaRegistries/General/pull/4819

```julia
julia> include("$(homedir())/.julia/dev/DiffEqGPU/test/runtests.jl")
 35.378608 seconds (84.52 M allocations: 4.226 GiB, 4.70% gc time)
 10.441867 seconds (25.32 M allocations: 1.265 GiB, 5.01% gc time)
  0.463463 seconds (1.57 M allocations: 89.961 MiB, 1.55% gc time)
  3.378792 seconds (12.41 M allocations: 560.900 MiB, 5.37% gc time)
  2.724377 seconds (9.25 M allocations: 434.991 MiB, 5.23% gc time)
  0.146231 seconds (221.40 k allocations: 11.530 MiB)
  8.088839 seconds (24.77 M allocations: 1.230 GiB, 7.18% gc time)
 14.162836 seconds (37.52 M allocations: 1.781 GiB, 4.23% gc time)
  3.985191 seconds (10.85 M allocations: 607.187 MiB, 4.59% gc time)
  4.521510 seconds (10.67 M allocations: 519.446 MiB, 3.03% gc time)
 16.842116 seconds (35.60 M allocations: 2.118 GiB, 4.36% gc time)
  9.553006 seconds (32.93 M allocations: 1.735 GiB, 6.00% gc time)
EnsembleSolution Solution of length 10 with uType:
ODESolution{Float32,2,Array{Array{Float32,1},1},Nothing,Nothing,Array{Float32,1},Array{Array{Array{Float32,1},1},1},ODEProblem{Array{Float32,1},Tuple{Float32,Float32},true,Array{Float32,1},ODEFunction{true,typeof(lorenz),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},DiffEqBase.StandardODEProblem},TRBDF2{0,true,DefaultLinSolve,NLNewton{Rational{Int64},Rational{Int64},Rational{Int64}},DataType},OrdinaryDiffEq.InterpolationData{ODEFunction{true,typeof(lorenz),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Array{Array{Float32,1},1},Array{Float32,1},Array{Array{Array{Float32,1},1},1},OrdinaryDiffEq.TRBDF2Cache{Array{Float32,1},Array{Float32,1},Array{Float32,1},OrdinaryDiffEq.TRBDF2Tableau{Float32,Float32},OrdinaryDiffEq.NLSolver{NLNewton{Rational{Int64},Rational{Int64},Rational{Int64}},true,Array{Float32,1},Float32,Float32,OrdinaryDiffEq.NLNewtonCache{Array{Float32,1},Float32,Array{Float32,1},Array{Float32,1},Array{Float32,2},Array{Float32,2},Array{Float32,1},DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lorenz),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float32,Array{Float32,1}},ForwardDiff.JacobianConfig{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lorenz),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float32,Array{Float32,1}},Float32},Float32,3,Tuple{Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lorenz),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float32,Array{Float32,1}},Float32},Float32,3},1},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.UJacobianWrapper{ODEFunction{true,typeof(lorenz),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float32,Array{Float32,1}},Float32},Float32,3},1}}},DefaultLinSolve,Float32}}}},DiffEqBase.DEStats}
```

Note that `] test DiffEqGPU` fails to compile. That might be due to bounds check.

**Comment by ChrisRackauckas** [comment]
Alright. And I think someone needs to manually restart the CI computer

**Pull request merged**